packages/playwright-core/src/client/network.ts

```diff
@@ -342,7 +342,7 @@ export class Route extends ChannelOwner<channels.RouteChannel> implements api.Ro
 
     const headers: Headers = {};
     for (const header of Object.keys(headersOption || {}))
-      headers[header.toLowerCase()] = String(headersOption![header]);
+      headers[header.toLowerCase()] = Array.isArray(headersOption![header]) ? headersOption![header] : String(headersOption![header]);
```

Comment:
That's not how multi-value headers are concatenated. Please start with filing a bug and we'll help you navigate it!